pkg/ingester/ingester.go

```diff
@@ -853,9 +859,12 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, pushReq *push.Request) e
 		stats pushStats
 
 		firstPartialErr    error
-		updateFirstPartial = func(errFn func() error) {
+		updateFirstPartial = func(sampler *util_log.Sampler, errFn func() error) {
 			if firstPartialErr == nil {
 				firstPartialErr = errFn()
+				if sampler != nil {
```

Comment:
This was suggested in combination of making `errFn` return only `ingesterError`, and changing `firstPartialErr`  to that type as well. It may make more sense then, but we can keep it for now.